plugins/inputs/linux_cpu/linux_cpu_test.go

```diff
@@ -0,0 +1,98 @@
+//go:build linux
+
+package linux_cpu
+
+import (
+	"github.com/influxdata/telegraf/testutil"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestNoMetrics(t *testing.T) {
+	plugin := &LinuxCPU{}
+	require.Error(t, plugin.Init())
+}
+
+func TestNoCPUs(t *testing.T) {
+	td := t.TempDir()
+
+	plugin := &LinuxCPU{
+		Metrics:   []string{"cpufreq"},
+		PathSysfs: td,
+	}
+	require.Error(t, plugin.Init())
+}
+
+func TestGatherCPUFreq(t *testing.T) {
+	td := t.TempDir()
+
+	require.NoError(t, os.MkdirAll(td+"/devices/system/cpu/cpu0/cpufreq", os.ModePerm))
+	require.NoError(t, os.WriteFile(td+"/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq", []byte("250\n"), 0644))
+	require.NoError(t, os.WriteFile(td+"/devices/system/cpu/cpu0/cpufreq/scaling_min_freq", []byte("100\n"), 0644))
+	require.NoError(t, os.WriteFile(td+"/devices/system/cpu/cpu0/cpufreq/scaling_max_freq", []byte("255\n"), 0644))
+
+	require.NoError(t, os.MkdirAll(td+"/devices/system/cpu/cpu1/cpufreq", os.ModePerm))
+	require.NoError(t, os.WriteFile(td+"/devices/system/cpu/cpu1/cpufreq/scaling_cur_freq", []byte("123\n"), 0644))
+	require.NoError(t, os.WriteFile(td+"/devices/system/cpu/cpu1/cpufreq/scaling_min_freq", []byte("80\n"), 0644))
+	require.NoError(t, os.WriteFile(td+"/devices/system/cpu/cpu1/cpufreq/scaling_max_freq", []byte("230\n"), 0644))
```

Comment:
Would be nice to test scenario when property cannot be loaded (property path doesn't exist and/or value of property cannot be parsed to uint)